src/frequenz/client/common/metric/__init__.py

```diff
@@ -0,0 +1,141 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Module to define the metrics used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
```

Comment:
This is not needed if you use `Self`:

```suggestion
from typing import Self

```